server/Serials/Model/User.hs

```diff
@@ -0,0 +1,134 @@
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE OverloadedStrings #-}
+
+module Serials.Model.User where
+
+import Prelude hiding (id)
+
+import Control.Applicative
+
+import Data.Text (Text, unpack, pack)
+import Data.ByteString.UTF8 (fromString, toString)
+import Data.Aeson (ToJSON, FromJSON, Value(..), toJSON, parseJSON, object, (.=), (.:), (.:?))
+import Control.Monad.IO.Class (liftIO)
+import Control.Monad (mzero)
+import Data.Maybe (catMaybes, fromJust, listToMaybe)
+import Data.Pool
+import Data.Time
+import Crypto.BCrypt
+
+import GHC.Generics
+import qualified Database.RethinkDB.NoClash as R
+import Database.RethinkDB.NoClash hiding (table, Object)
+
+import Serials.Model.UserSignupFields (UserSignupFields)
+import qualified Serials.Model.UserSignupFields as U
+import Serials.Lib.Crud
+import Serials.Lib.Helpers
+
+data User = User {
+  id :: Text
+  , firstName :: Text
+  , lastName :: Text
+  , email :: Text
+
+  , hashedPassword :: Maybe Text
+
+  , token :: Text
+  , admin :: Bool -- Should we have more permissions?
+
+  , created :: UTCTime
+} deriving (Show, Generic)
+
+instance FromJSON User where
+  parseJSON (Object v) = User
```

Comment:
Cool. It may be easier to do this with composed types: `data SecureUser = SecureUser { user :: User, extra :: ... }`